src/signatures.rs

```diff
@@ -0,0 +1,988 @@
+pub mod system {
```

Comment:
Any particular reason to put all these into a separate `signatures` module? Why not keep them close by to their definitions? That is, just move each one above the respective `pub struct Protocol {`? This would keep the code-structure closer to the specification and thus make it easier to review new additions or audit existing protocols.

We generally try to stay as close to the structure of the specification as possible. So I was just wondering what your rationale was.

Reply:
Well, there was not any big reason to keep signatures in a separate module. I just thought that I should probably not clutter other namespaces in the generated docs.

Reply:
@dvdhrm Sure, if you don't mind then feel free to move them to whichever place makes the most sense to you. 

I am a bit preoccupied with getting UEFI std to build again since one of the Rust recent commits breaks all the `restricted_std` targets.